biz.aQute.resolve/src/biz/aQute/resolve/ProjectResolver.java

```diff
@@ -81,10 +82,14 @@ private String toString(ServiceReference< ? > sr) {
 	private ResolveProcess					resolve		= new ResolveProcess();
 	private Collection<ResolutionCallback>	cbs			= new ArrayList<ResolutionCallback>();
 
-	public ProjectResolver(Project project) {
+	public ProjectResolver(Project project) throws Exception {
 		super(project);
 		getSettings(project);
 		this.project = project;
+
+		Workspace workspace = project.getWorkspace();
+		if (workspace != null)
+			project.addBasicPlugin(new WorkspaceResourcesRepository(workspace));
```

Comment:
Also, you did not modify the bnd resolve operation to remove its creation of a workspace repository since it is now also done by the ProjectResolver constructor.